components/expenses/ExpenseFormPayeeStep.js

```diff
@@ -136,6 +136,12 @@ const ExpenseFormPayeeStep = ({
     __background__: 'white',
   });
 
+  useEffect(() => {
+    setLocationFromPayee(formik, collective.host);
```

Comment:
There's a misconception here, we're not falling back to the collective's host address, we're falling back to the payee host address.